core/services/subscription.go

```diff
@@ -117,175 +115,77 @@ func NewInitiatorSubscription(
-// GetJobSpecID method
-func ReceiveLogRequest(runManager RunManager, le models.LogRequest) {
-	if !le.Validate() {
-		logger.Debugw("discarding INVALID EVENT LOG", "log", le.GetLog())
-		return
-	}
-
-	if le.GetLog().Removed {
-		logger.Debugw("Skipping run for removed log", "log", le.GetLog(), "jobId", le.GetJobSpecID().String())
-		return
-	}
-
-	le.ToDebug()
-
-	runJob(runManager, le)
-
-}
-
-func runJob(runManager RunManager, le models.LogRequest) {
-	jobSpecID := le.GetJobSpecID()
-	initiator := le.GetInitiator()
-
-	if err := le.ValidateRequester(); err != nil {
-		if _, e := runManager.CreateErrored(jobSpecID, initiator, err); e != nil {
-			logger.Errorw(e.Error())
-		}
-		logger.Errorw(err.Error(), le.ForLogger()...)
-		return
-	}
-
-	rr, err := le.RunRequest()
-	if err != nil {
-		if _, e := runManager.CreateErrored(jobSpecID, initiator, err); e != nil {
-			logger.Errorw(e.Error())
-		}
-		logger.Errorw(err.Error(), le.ForLogger()...)
-		return
-	}
-
-	_, err = runManager.Create(jobSpecID, &initiator, le.BlockNumber(), &rr)
-	if err != nil {
-		logger.Errorw(err.Error(), le.ForLogger()...)
-	}
-}
-
-// ManagedSubscription encapsulates the connecting, backfilling, and clean up of an
-// ethereum node subscription.
-type ManagedSubscription struct {
-	done              chan struct{}
-	logSubscriber     eth.Client
-	logs              chan types.Log
-	ethSubscription   ethereum.Subscription
-	callback          func(types.Log)
-	backfillBatchSize uint32
-}
-
-// NewManagedSubscription subscribes to the ethereum node with the passed filter
-// and delegates incoming logs to callback.
-func NewManagedSubscription(logSubscriber eth.Client, filter ethereum.FilterQuery, callback func(types.Log), backfillBatchSize uint32) (*ManagedSubscription, error) {
-	ctx := context.Background()
+) (*InitiatorSubscription, error) {
 	logs := make(chan types.Log)
-	done := make(chan struct{})
-	es, err := logSubscriber.SubscribeFilterLogs(ctx, filter, logs)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	es, err := client.SubscribeFilterLogs(ctx, filter, logs)
 	if err != nil {
 		return nil, err
 	}
-
-	sub := ManagedSubscription{
-		done:              done,
-		logSubscriber:     logSubscriber,
-		callback:          callback,
-		logs:              logs,
+	return &InitiatorSubscription{
+		Initiator:         initr,
+		done:              make(chan struct{}),
+		logSubscriber:     client,
 		ethSubscription:   es,
+		filter:            filter,
+		logs:              logs,
+		runManager:        runManager,
 		backfillBatchSize: backfillBatchSize,
-	}
-	go sub.listenToLogs(filter)
-	return &sub, nil
+		callback:          callback,
+	}, nil
 }
 
-// Unsubscribe closes channels and cleans up resources.
-func (sub *ManagedSubscription) Unsubscribe() {
-	if sub.ethSubscription != nil {
-		timedUnsubscribe(sub.ethSubscription)
-	}
-	close(sub.done)
+func (sub *InitiatorSubscription) Start() {
```

Comment:
Pull out a separate Start function to separate the setup from the beginning of a long lived goroutine